File: common/src/main/java/de/werkelmann/rekote/address/AddressChecker.java
package de.werkelmann.rekote.address;

import java.util.LinkedList;
import java.util.List;

public abstract class AddressChecker {

    protected List<AddressRule> rules = new LinkedList<>();

    public boolean check(String address) {
        for (AddressRule rule : rules) {
            if (!rule.isValid(address)) {
                return false;
            }
        }
        return true;
    }
}

File: common/src/main/java/de/werkelmann/rekote/address/AddressChecker.kt
package de.werkelmann.rekote.address

import de.werkelmann.rekote.address.ip.IpChecker
import de.werkelmann.rekote.address.url.UrlChecker

class AddressChecker : ServerConnectionChecker() {

    val urlChecker = UrlChecker()
    val ipChecker = IpChecker()

    override fun check(address: String?): Boolean {
        return urlChecker.check(address) || ipChecker.check(address)
    }
}


